flixel/input/actions/FlxActionInputAnalog.hx

```diff
@@ -0,0 +1,367 @@
+	var STOPPED      =  cast FlxInputState.RELEASED;		// is 0
+	var MOVED        =  cast FlxInputState.PRESSED;			// is !0
+	var JUST_MOVED   =  cast FlxInputState.JUST_PRESSED;	// became !0 on this frame
+}
+
+class FlxActionInputAnalogClickAndDragMouseMotion extends FlxActionInputAnalogMouseMotion
+{
+	private var button:FlxMouseButtonID;
+	
+	/**
+	 * Mouse input -- same as FlxActionInputAnalogMouseMotion, but requires a particular mouse button to be PRESSED
+	 * Very useful for e.g. panning a map or canvas around
+	 * @param	ButtonID	Button identifier (FlxMouseButtonID.LEFT / MIDDLE / RIGHT)
+	 * @param	Trigger	What state triggers this action (MOVED, JUST_MOVED, STOPPED, JUST_STOPPED)
+	 * @param	Axis	which axes to monitor for triggering: X, Y, EITHER, or BOTH
+	 * @param	PixelsPerUnit	How many pixels of movement = 1.0 in analog motion (lower: more sensitive, higher: less sensitive)
+	 * @param	DeadZone	Minimum analog value before motion will be reported
+	 * @param	InvertY	Invert the Y axis
+	 * @param	InvertX	Invert the X axis
+	 */
+	public function new(ButtonID:FlxMouseButtonID, Trigger:FlxAnalogState, Axis:FlxAnalogAxis = EITHER, PixelsPerUnit:Int = 10, DeadZone:Float = 0.1, InvertY:Bool = false, InvertX:Bool = false)
+	{
+		super(Trigger, Axis, PixelsPerUnit, DeadZone, InvertY, InvertX);
+		button = ButtonID;
+	}
+	
+	override function updateVals(X:Float, Y:Float):Void 
+	{
+		var pass = false;
+		#if !FLX_NO_MOUSE
+		pass = switch (button)
+		{
+			case FlxMouseButtonID.LEFT: FlxG.mouse.pressed;
+			case FlxMouseButtonID.RIGHT: FlxG.mouse.pressedRight;
+			case FlxMouseButtonID.MIDDLE: FlxG.mouse.pressedMiddle;
+		}
+		#end
+		if (!pass)
+		{
+			X = 0;
+			Y = 0;
+		}
+		super.updateVals(X, Y);
+		
+	}
+}
+
+class FlxActionInputAnalogMouseMotion extends FlxActionInputAnalog
+{
+	private var lastX:Float = 0;
+	private var lastY:Float = 0;
+	private var pixelsPerUnit:Int;
+	private var deadZone:Float;
+	private var invertX:Bool;
+	private var invertY:Bool;
+	
+	/**
+	 * Mouse input -- X/Y is the RELATIVE motion of the mouse since the last frame
+	 * @param	Trigger	What state triggers this action (MOVED, JUST_MOVED, STOPPED, JUST_STOPPED)
+	 * @param	Axis	which axes to monitor for triggering: X, Y, EITHER, or BOTH
+	 * @param	PixelsPerUnit	How many pixels of movement = 1.0 in analog motion (lower: more sensitive, higher: less sensitive)
+	 * @param	DeadZone	Minimum analog value before motion will be reported
+	 * @param	InvertY	Invert the Y axis
+	 * @param	InvertX	Invert the X axis
+	 */
+	public function new(Trigger:FlxAnalogState, Axis:FlxAnalogAxis = EITHER, PixelsPerUnit:Int = 10, DeadZone:Float = 0.1, InvertY:Bool = false, InvertX:Bool = false)
+	{
+		pixelsPerUnit = PixelsPerUnit;
+		if (pixelsPerUnit < 1)
+			pixelsPerUnit = 1;
+		deadZone = DeadZone;
+		invertX = InvertX;
+		invertY = InvertY;
+		super(FlxInputDevice.MOUSE, -1, cast Trigger, Axis);
+	}
+	
+	override public function destroy():Void 
```

Comment:
if you just call `super.destroy()` here, then there is no need to override `destroy()`